src/arrayserialize.rs

```diff
@@ -24,20 +24,24 @@ impl<A, D, S> Serialize for ArrayBase<S, D>
         let mut struct_state = try!(serializer.serialize_struct("Array", 3));
         try!(serializer.serialize_struct_elt(&mut struct_state, "v", ARRAY_FORMAT_VERSION));
         try!(serializer.serialize_struct_elt(&mut struct_state, "dim", self.dim()));
-        try!(serializer.serialize_struct_elt(&mut struct_state, "data", self.iter()));
+        try!(serializer.serialize_struct_elt(&mut struct_state, "data", Sequence(self.iter())));
```

Comment:
Export nothing more publically than needed. A serialize impl for the iterator would be public.